core/leadership/interface.go

```diff
@@ -122,7 +122,7 @@ type Checker interface {
 	// This method returns a token that accepts a *[]txn.Op, into which
 	// it will (on success) copy mgo/txn operations that can be used to
 	// verify the unit's continued leadership as part of another txn.
-	LeadershipCheck(applicationName, unitName string) Token
+	LeadershipCheck(applicationId, unitId string) Token
```

Comment:
It's what you get from `someTag.Id()`. The "Name" in the original signature is misleading and caused me to waste quite a bit of time figuring out why leadership checks were failing in my tests (see the `leadership.Claimer` interface for comparison that uses Id)

Reply:
interesting.  I'll have to check.  I would have thought application id and application name were the same.  though storage id appears to be different.

Reply:
tag.Id() returns name for application, as it does for unit and other things like user
so appTag.Id() and appTag.Name() would be the same (same for unit)

Reply:
It's fine for applications but the unit one (which caused the issue in the first place) is not the same because of [this](https://github.com/juju/names/blob/v3/unit.go#L33) little transformation (last '-' to '/'). Took a while to figure out that this was the reason the leadership check was failing for my tests